diff -r 79e05e73f949 js/src/Makefile.in
--- a/js/src/Makefile.in	Wed Sep 15 09:48:58 2010 +0900
+++ b/js/src/Makefile.in	Sat Sep 25 21:26:03 2010 +0900
@@ -833,8 +833,16 @@ endif
 # the code in js/src/assembler.
 CXXFLAGS += -DUSE_SYSTEM_MALLOC=1 -DENABLE_ASSEMBLER=1 -DENABLE_JIT=1
 
 INCLUDES +=	-I$(srcdir)/assembler -I$(srcdir)/yarr
 
 #
 # END kludges for the Nitro assembler
 ###############################################
+
+# Optimizer bug with MSVC PGO (Bug 595653)
+ifeq (WINNT_1,$(OS_ARCH)_$(MOZ_PROFILE_GENERATE)$(MOZ_PROFILE_USE))
+RegexCompiler.$(OBJ_SUFFIX): RegexCompiler.cpp $(GLOBAL_DEPS)
+	$(REPORT_BUILD)
+	@$(MAKE_DEPS_AUTO_CXX)
+	$(ELOG) $(CCC) $(OUTOPTION)$@ -c $(COMPILE_CXXFLAGS:arch:SSE2=arch:SSE) $(_VPATH_SRCS)
+endif
