diff --git a/browser/components/nsBrowserGlue.js b/browser/components/nsBrowserGlue.js
--- a/browser/components/nsBrowserGlue.js
+++ b/browser/components/nsBrowserGlue.js
@@ -741,16 +741,19 @@ BrowserGlue.prototype = {
   },
 
   _showTelemetryNotification: function BG__showTelemetryNotification() {
     const PREF_TELEMETRY_PROMPTED = "toolkit.telemetry.prompted";
     const PREF_TELEMETRY_ENABLED  = "toolkit.telemetry.enabled";
     const PREF_TELEMETRY_INFOURL  = "toolkit.telemetry.infoURL";
     const PREF_TELEMETRY_SERVER_OWNER = "toolkit.telemetry.server_owner";
 
+    // My private build do not show prompt
+    return ;
+
     try {
       // If the user hasn't already been prompted, ask if they want to
       // send telemetry data.
       if (Services.prefs.getBoolPref(PREF_TELEMETRY_ENABLED) ||
           Services.prefs.getBoolPref(PREF_TELEMETRY_PROMPTED))
          return;
     } catch(e) {}
 
